lib/reek/smells/smell_warning.rb

```diff
@@ -17,6 +17,8 @@ class SmellWarning
       attr_reader :context, :lines, :message, :parameters, :smell_detector, :source
       def_delegators :smell_detector, :smell_category, :smell_type
 
+      DETAIL_KEYS = %i(message lines context source parameters)
```

Comment:
We compare two things:

1.) "attributes": message lines context source parameters
and
2.) "parameters". This "attribute" has "keys". So let's use "keys" only in the context of "parameters", but not in the context of "attributes"